lib/active_model/serializer/adapter/json.rb

```diff
@@ -43,6 +29,31 @@ def serializable_hash(options = nil)
           { root => @result }
         end
 
+        def serialize_array_without_root(serializer, options)
+          serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+        end
+
+        # TODO: what is a virtual value?
```

Comment:
it's an internal thing that uses that value instead of a serialized attribute.  e.g. there's no serializer for Ruby arrays, so it would just use the array as the virtual value. see https://github.com/rails-api/active_model_serializers/pull/962